src/ol/format/IIIFInfo.js

```diff
@@ -0,0 +1,368 @@
+    }
+    // TODO potentially dangerous
+    attributions.push(license);
+  }
+  return {
+    url: iiifInfo.imageInfo['@id'].replace(/\/?(info.json)?$/g, ''),
+    sizes: iiifInfo.imageInfo.sizes === undefined ? undefined : iiifInfo.imageInfo.sizes.map(function(size) {
+      return [size.width, size.height];
+    }),
+    tileSize: iiifInfo.imageInfo.tiles === undefined ? undefined : [
+      iiifInfo.imageInfo.tiles.map(function(tile) {
+        return tile.width;
+      })[0],
+      iiifInfo.imageInfo.tiles.map(function(tile) {
+        return tile.height === undefined ? tile.width : tile.height;
+      })[0]
+    ],
+    resolutions: iiifInfo.imageInfo.tiles === undefined ? undefined :
+      iiifInfo.imageInfo.tiles.map(function(tile) {
+        return tile.scaleFactors;
+      })[0],
+    supports: [...levelProfile.supports, ...profileSupports],
+    formats: [...levelProfile.formats, ...profileFormats],
+    qualities: [...levelProfile.qualities, ...profileQualities],
+    attributions: attributions.length == 0 ? undefined : attributions
+  };
+}
+
+function generateVersion3Options(iiifInfo) {
+  const levelProfile = iiifInfo.getComplianceLevelSupportedFeatures();
+  return {
+    url: iiifInfo.imageInfo['id'],
+    sizes: iiifInfo.imageInfo.sizes === undefined ? undefined : iiifInfo.imageInfo.sizes.map(function(size) {
+      return [size.width, size.height];
+    }),
+    tileSize: iiifInfo.imageInfo.tiles === undefined ? undefined : [
+      iiifInfo.imageInfo.tiles.map(function(tile) {
+        return tile.width;
+      })[0],
+      iiifInfo.imageInfo.tiles.map(function(tile) {
+        return tile.height;
+      })[0]
+    ],
+    resolutions: iiifInfo.imageInfo.tiles === undefined ? undefined :
+      iiifInfo.imageInfo.tiles.map(function(tile) {
+        return tile.scaleFactors;
+      })[0],
+    supports: iiifInfo.imageInfo.extraFeatures === undefined ? levelProfile.supports :
+      [...levelProfile.supports, ...iiifInfo.imageInfo.extraFeatures],
+    formats: iiifInfo.imageInfo.extraFormats === undefined ? levelProfile.formats :
+      [...levelProfile.formats, ...iiifInfo.imageInfo.extraFormats],
+    qualities: iiifInfo.imageInfo.extraQualities === undefined ? levelProfile.qualities :
+      [...levelProfile.supports, ...iiifInfo.imageInfo.extraQualities],
+    maxWidth: undefined,
+    maxHeight: undefined,
+    maxArea: undefined,
+    attributions: undefined
+  };
+}
+
+const versionFunctions = {};
+versionFunctions[Versions.VERSION1] = generateVersion1Options;
+versionFunctions[Versions.VERSION2] = generateVersion2Options;
+versionFunctions[Versions.VERSION3] = generateVersion3Options;
+
+/**
+ * @classdesc
+ * Format for transforming IIIF Image API image information responses into
+ * IIIF tile source ready options
+ *
+ * @api
+ */
+class IIIFInfo {
+
+  /**
+   * @param {Object|string} imageInfo Deserialized image information JSON response
+   * object or JSON response as string
```

Comment:
Would it be possible to provide a typedef for this object?

Reply:
That would require three different typedefs as each IIIF Image API info.json has a different structure. I'll look into it.

Reply:
I added typedefs for all possible objects as well as their dependent objects.
One problem: IIIF Image information responses from version 1.1 on are JSON-LD. And I'm not sure how to reference `@id` or `@context` in the doc, e.g.:

https://github.com/lutzhelm/openlayers/blob/a2c1e9cb5bc48d63f6bd00ea0299394ec162f237/src/ol/format/IIIFInfo.js#L38-L39

Reply:
I'm curious how these will be handled as well.  If you rebase your branch on the latest in master (`git pull --rebase openlayers master`), we'll get the API docs built and hosted in the CI job.

You can build the docs locally with `npm run apidoc` (the output will be in `build/apidoc`).

I'm also curious if `tsc` complains about those typedefs (`npm run typecheck` to find out).

Reply:
`npm run apidoc` worked once I fixed another issue.
`tsc` complains about every contained JSON-LD keyword: `Identifier expected.`